servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.h

```diff
@@ -691,15 +692,39 @@ class RenderForwardClustered : public RendererSceneRenderRD {
 			sorter.sort(elements.ptr(), elements.size());
 		}
 
-		struct SortByReverseDepthAndPriority {
+		struct StandardSceneSort {
 			_FORCE_INLINE_ bool operator()(const GeometryInstanceSurfaceDataCache *A, const GeometryInstanceSurfaceDataCache *B) const {
-				return (A->sort.priority == B->sort.priority) ? (A->owner->depth > B->owner->depth) : (A->sort.priority < B->sort.priority);
+				if (A->sort.priority == B->sort.priority) {
+					if (Math::is_equal_approx(A->owner->depth, B->owner->depth)) {
+						if (A->owner->sorting_stacked_order == B->owner->sorting_stacked_order) {
+							if (A->sort.surface_index == B->sort.surface_index) {
+								if (A->sort.material_depth == B->sort.material_depth) {
+									return (A->sort.geometry_id < B->sort.geometry_id);
+								}
+								return (A->sort.material_depth < B->sort.material_depth);
```

Comment:
At this point we should be using a sorting struct like for the opaque pass. This code is getting a bit unwieldy and I am worried about the performance implications of extracting each element from the sort struct. 

Keep in mind, when you access an element like this, the compiler will emit a bit shift and a mask instruction for each element access and it adds up really quickly.

Reply:
I am working on a more performant version now giving a custom sort key for transparent sorting that would be usesd after the depth